src/irmin-pack/irmin_pack.ml

```diff
@@ -399,6 +399,51 @@ struct
     end
   end
 
+  let integrity_check ppf (t : X.Repo.t) =
+    Fmt.pf ppf "running the integrity check\n%!";
+    let commits = ref 0 in
+    let contents = ref 0 in
+    let nodes = ref 0 in
+    let pp_stats ppf () =
+      Fmt.pf ppf "%4dk blobs / %4dk trees / %4dk commits" (!contents / 1000)
+        (!nodes / 1000) (!commits / 1000)
+    in
+    let pr_stats () = Fmt.epr "\r%a%!" pp_stats () in
+    let count_increment count =
+      incr count;
+      if !count mod 100 = 0 then pr_stats ()
+    in
+    Index.iter
+      (fun k (offset, length, m) ->
+        match m with
+        | 'B' ->
+            Lwt.async (fun () ->
```

Comment:
I guess the `Lwt.async` calls are not really necessary anymore as the integrity callbacks are not in the lwt monad.